File: Prog4/src/Queue.java
/**
This method controls the queue for aircraft. 
*/

/**
 
 @author Ryan Reistroffer, Steven Storkson
 */
public class Queue <E>
{
   private E [] elements;
   private int front, rear, count;
   
   /**
   Constructor for elements of queue
   @param size 
   */
   public Queue(int size)
   {
      elements = (E[]) new Object[size];
      front = rear = count = 0;
   }
   
   /**
   Brings in an Object and assigns it to the current array value of rear.
   Queue is circular.
   @param x 
   */
   public void enqueue( Object x)
   {
      elements[rear] = (E) x;
      rear = ( rear + 1 ) % elements.length;
      count++;
   }
   
   /**
   Returns the first object in the queue. The queue is circular.
   @return x: first object in queue
   */
   public Object dequeue( )
   {
      Object x = elements[front];
      front = ( front + 1 ) % elements.length;
      count--;
      return x;
   }
   
   /**
   Checks the queue to see if it is empty
   @return 
   */
   public boolean isEmpty()
   {
      return (count == 0);
   }
   
   /**
   Will clear the queue array.
   */
   public void clearQueue()
   {
      front = rear = count = 0;
   }
   
   /**
   Checks to see if the queue is full
   @return true if full
   */
   public boolean isFull()
   {
      return count == elements.length;
   }
   
   /**
   Returns the size of the queue array
   @return count
   */
   public int size()
   {
      return count;
   }
   
}

File: OOPS2 - C++/prog1/Queue.cpp
//---------------------------------------------------------------------
// Name:    Steven Storkson
// Project: Reverse Polish Notation Evaluator with Queue and Stack.
// Purpose: The implementation of the Queue class methods.  It has a 
//          default constructor for initialization and methods for adding
//          and removing from the queue.  It also is able to check if the 
//          queue is full or empty and can return the current size of the 
//          queue.
//---------------------------------------------------------------------
#include "Queue.h"

Queue::Queue() 
{
   front = rear =  count = 0;
}

void Queue::enqueue(QInfoType x) 
{
   elements[rear] = x;
   rear = (rear + 1) % MAXQ;
   count++;
}

QInfoType Queue::dequeue() 
{
   QInfoType x = elements[front];
   front = (front + 1) % MAXQ;
   count--;
   return x;
}

bool Queue::isEmpty() 
{
   if(front == rear) 
      return true;
   else
      return false;
}

bool Queue::isFull() 
{
   if (front == (rear + 1) % MAXQ)
      return true;
   else
      return false;
}


void Queue::clear() 
{
   front = rear = count = 0;
}

int Queue::size() 
{
   return count;
}

#ifdef TESTING_Q

// ------------------------------
// Testbed main
// ------------------------------
int main()
{
   Queue q;
   q.enqueue(12.1);
   cout << q.dequeue() << endl;
   q.enqueue(13.5);
   q.enqueue(5.9);
   if (q.isEmpty()) 
      cout << "Queue is empty" << endl;
   else
      cout << "Queue is Not Empty." << endl;
   
   if (q.isFull()) 
      cout << "Queue is full" << endl;
   else
      cout << "Queue is Not full." << endl;
   cout << q.size() << " Is how many elements I have in my Queue." << endl;
   cout << q.dequeue() << " I should have one more element in the queue. "<< endl;
   q.clear();
   cout << "I just cleared the queue." << endl;
   
   
   // .... etc, to test all the methods.
   return 0;
}

#endif
